packages/js/product-editor/src/blocks/shipping-dimensions/edit.tsx

```diff
@@ -0,0 +1,247 @@
+/**
+ * External dependencies
+ */
+import { useBlockProps } from '@wordpress/block-editor';
+import { BlockEditProps } from '@wordpress/blocks';
+import { OPTIONS_STORE_NAME, ProductDimensions } from '@woocommerce/data';
+import { useInstanceId } from '@wordpress/compose';
+import { useEntityProp } from '@wordpress/core-data';
+import { useSelect } from '@wordpress/data';
+import {
+	createElement,
+	createInterpolateElement,
+	useState,
+} from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import classNames from 'classnames';
+import {
+	BaseControl,
+	// @ts-expect-error `__experimentalInputControl` does exist.
+	__experimentalInputControl as InputControl,
+} from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { ShippingDimensionsBlockAttributes } from './types';
+import { useProductHelper } from '../../hooks/use-product-helper';
+import {
+	HighlightSides,
+	ShippingDimensionsImage,
+} from '../../components/shipping-dimensions-image';
+import { useValidation } from '../../hooks/use-validation';
+
+export function Edit( {}: BlockEditProps< ShippingDimensionsBlockAttributes > ) {
+	const blockProps = useBlockProps( {
+		className: 'wp-block-woocommerce-product-shipping-dimensions-fields',
```

Comment:
Any reason we need to manually override the class name here?

Reply:
Just to wrap any block only style. If you have another idea for that, please let know.

Reply:
I don't think it overrides anything here, just [adds the class to the other block classes](https://github.com/WordPress/gutenberg/blob/917afa465ba45d6fb7276bd9833c099cb3454c46/packages/block-editor/src/components/block-list/use-block-props/index.js#L161).

Reply:
I don't think it overrides anything here, just [adds the class to the other block classes](https://github.com/WordPress/gutenberg/blob/917afa465ba45d6fb7276bd9833c099cb3454c46/packages/block-editor/src/components/block-list/use-block-props/index.js#L168).

Reply:
I should specify that it does not override the block classes, but because this is a duplicate class that already exists for the block, it's overriding the original one.  We can remove it or use a different class name and `wp-block-woocommerce-product-shipping-dimensions-fields` will still get added:

<img width="373" alt="Screen Shot 2023-04-13 at 6 39 46 PM" src="https://user-images.githubusercontent.com/10561050/231919910-6f0676c0-8396-4123-a18d-53dc6831b814.png">
(Passed `this-is-the-props-class` for the `className` property in the above, note that the block class still exists)


Here's where the default block class gets added: 
 https://github.com/WordPress/gutenberg/blob/917afa465ba45d6fb7276bd9833c099cb3454c46/packages/block-editor/src/components/block-list/use-block-props/index.js#L171

Reply:
Ohh!!! I see, the block className is autogenerated base on the block name. I never understood you from the very beginning. Well thanks now it was solved -> https://github.com/woocommerce/woocommerce/pull/37683/commits/7e54f935520fcb5cdf96c1c3e2f2092a8a3eaa32
cc @joelclimbsthings